modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -85,5 +85,30 @@ def report msg, response, error_only
       logger.error "Dnscmd failed:\n" + (response.is_a?(Array) ? response.join("\n") : "Response was not an array! #{response}")
       raise Proxy::Dns::Error.new("Unknown error while processing '#{msg}'")
     end
+
+    def match_zone(fqdn)
+      dns_zones = ::Proxy::Dns::Plugin.settings.dns_zones || nil
+      found_match = false
+      weight = 0 # sub zones might be independent from similar named parent zones; use weight for longest suffix match
+      matched_zone = fqdn.sub(/[^.]+./,'')
+
+      if dns_zones.is_a?(Array)
+        dns_zones.each do |zone|
+          zone_labels = zone.split(".").reverse
+          zone_weight = zone_labels.length
+          fqdn_labels = fqdn.split(".")
+          fqdn_labels.shift
+          is_match = zone_labels.all? { |zone_label| zone_label == fqdn_labels.pop }
+          # match only the longest zone suffix
+          (matched_zone, weight, found_match = zone, zone_weight, true) if is_match && zone_weight >= weight
```

Comment:
@witlessbird Certainly. Added `test_dns_zone_matching`. I've opted for a separate test rather than extending the existing tests; tell me what you like this better.

Right now I have trouble with settings. They get ignored. So far I tried changing [this](https://github.com/theforeman/smart-proxy/blob/develop/test/dns_dnscmd/dnscmd_test.rb#L7):
```ruby
Proxy::Dns::Dnscmd::Plugin.load_test_settings(
        :dns_server => 'a_server',
        :dns_zones => ['ad.domain.com', 'domain.com', '33.168.192.in-addr.arpa'])
```